api/src/main/java/org/pmiops/workbench/actionaudit/bucket/BucketAuditQueryServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+package org.pmiops.workbench.actionaudit.bucket;
+
+import static org.pmiops.workbench.exfiltration.ObjectNameSizeConstants.THRESHOLD;
+
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.TableResult;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import javax.inject.Provider;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.config.WorkbenchConfig.BucketAuditConfig;
+import org.pmiops.workbench.utils.FieldValues;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BucketAuditQueryServiceImpl implements BucketAuditQueryService {
+
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+  private final BigQueryService bigQueryService;
+
+  private static final String QUERY =
```

Comment:
This query processes 171.38 GB when run in prod ~ $1, the job will run 4 times a day $4 a day, so the total spend would be ~$120 / month

Reply:
and in test ~17 GB so the cost would be $0.1, and I'll make it run once a day to reduce any associated costs.